frontend/include/chpl/types/LoopExprIteratorType.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2024 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_TYPES_LOOP_EXPR_ITERATOR_TYPE_H
+#define CHPL_TYPES_LOOP_EXPR_ITERATOR_TYPE_H
+
+#include "chpl/types/Type.h"
+#include "chpl/types/QualifiedType.h"
+#include "chpl/types/IteratorType.h"
+
+namespace chpl {
+namespace types {
+
+class LoopExprIteratorType final : public IteratorType {
+ private:
+  bool isZippered_;
+  bool supportsParallel_;
+  QualifiedType iterand_;
+  ID sourceLocation_;
```

Comment:
This one is interesting because historically we have not been able to have IDs for compiler-generated stuff. So if there came a time when we needed to resolve a compiler-generated loop expression, we would need to either change this or fabricate an ID.

Reply:
I can add a TODO / note, but I am not personally too worried by this field in the meantime.